cpp/src/io/orc/orc_gpu.hpp

```diff
@@ -39,6 +40,19 @@ namespace gpu {
 using cudf::detail::device_2dspan;
 using cudf::detail::host_2dspan;
 
+auto constexpr KEY_SENTINEL   = size_type{-1};
+auto constexpr VALUE_SENTINEL = size_type{-1};
+
+using map_type = cuco::static_map<size_type, size_type>;
+
+/**
+ * @brief The alias of `map_type::pair_atomic_type` class.
+ *
+ * Declare this struct by trivial subclassing instead of type aliasing so we can have forward
+ * declaration of this struct somewhere else.
+ */
+struct slot_type : public map_type::pair_atomic_type {};
```

Comment:
```suggestion
struct slot_type : public map_type::slot_type {};
```
Can we use [slot_type](https://github.com/NVIDIA/cuCollections/blob/303f134573afa315cf14fca3f7a0b730438497c3/include/cuco/static_map.cuh#L596) directly?

Reply:
Would you be opposed to removing the slot_type "alias"? Once I apply your suggestion, the benefit from the "alias" shrinks and it really only saves a few words.